File: RoRClient/Model/Models/Map.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RoRClient.Model.Models
{
    public class Map : ModelBase
    {
        //Map mit Squares gefüllt (Spielfeld)

        private const int mapSize = 25;
        public static int MapSize => mapSize;

        Square[,] squares;

        public Square[,] Squares
        {
            get
            {
                return squares;
            }
            set
            {
                squares = value;
            }
        }


        public Map () : base()
        {
            squares = new Square[mapSize, mapSize];
            InitSquares();
            // For testing purposes
            CreateRandomRailsForTest();
        }

        /// <summary>
        /// Jedes Square auf der Map braucht einen Index,
        /// um jedem Objekt, das auf einem Square platziert, ein eindeutiges Objekt zuzuordnen
        /// </summary>
        private void InitSquares()
        {
            for (int x = 0; x < mapSize; x++)
            {
                for (int y = 0; y < mapSize; y++)
                {
                    Square s = new Square(x, y);
                    squares[x, y] = s;
                }
            }
        }

        private void CreateRandomRailsForTest()
        {
            for (int x = 0; x < mapSize; x++)
            {
                for (int y = 0; y < mapSize; y++)
                {
                    Random rand = new Random();
                    Rail rail = new Rail(squares[x, y], new RailSection(RailSectionPosition.NORTH, RailSectionPosition.SOUTH));
                    squares[x, y].PlaceableOnSquare = rail;
                }
            }
        }

        public Square GetSquare (int x, int y)
        {
            //Getter für ein Square auf der Map
            return this.squares[x, y];
        }
    }
}

File: RoRServer/src/main/java/models/Game/Map.java
package models.Game;


public class Map {
	
	private static Map map = null;
	private Square squares [][];
	private final int mapSize = 3;
	
	private Map() {
		squares = new Square[mapSize][mapSize];
		
		for(int i= 0; i < mapSize; i++) {
			for(int j = 0; j < mapSize; j++) {
				Square s = new Square(i, j);
				squares[i][j] = s;
			}
		}
	}
	
	public static Map getInstance() {
		if (map == null) {
			map = new Map();
		}
		return map;
	}
	
	public Square getSquare(int i, int j) {
		return squares[i][j];
	}
	
	public Square[][] getSquares() {
		return squares;
	}
	
	public void setSquares(Square squares[][]) {
		this.squares = squares;
	}

}
